src/axom/core/memory_management.hpp

```diff
@@ -379,6 +393,43 @@ inline void copy(void* dst, const void* src, std::size_t numbytes) noexcept
 #endif
 }
 
+//------------------------------------------------------------------------------
+template <typename T>
+inline void fill(void* dst, std::size_t n, const T& value) noexcept
+{
+  bool doHostFill = true;
```

Comment:
The docs say that T is a "plain-old-datatype"

Should there be a `static_assert` to check that `T` satisfies this expectation?